src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Interop/InvokeDotNetMethodWithJsonMarshalling.ts

```diff
@@ -0,0 +1,183 @@
+import { platform } from '../Environment';
+import { System_String, Pointer } from '../Platform/Platform';
+import { getRegisteredFunction } from './RegisteredFunction';
+import { error } from 'util';
+
+export interface MethodOptions {
```

Comment:
Would `MethodIdentifier` be a clearer name? I think `MethodOptions` is a bit strange because there aren't really any "options" here other than just identifying a specific method.

Reply:
Update: I agree with Ryan's comments below that since `MethodInstance` never gets used separately from `MethodOptions`, these could be combined into a single type (perhaps named `MethodIdentifier`, or `MethodInstance` if you prefer).

Reply:
MethodOptions is the container, method instance holds the information for the specific method instantiation.